test/models/sync_plan_test.rb

```diff
@@ -129,5 +129,32 @@ def test_schedule_format_disabled
       assert_match(/\/PT24H$/, schedule)
       assert_includes schedule, @plan.sync_date.iso8601
     end
+
+    def test_add_product
+      product = katello_products(:redhat)
+      @plan.products << product
+      @plan.save!
```

Comment:
Same thing with these tests. If you don't need to test creation/deletion but just validation, try using the `valid?` method, which runs the validations but doesn't save to the database.